sdk/src/main/java/com/bugsnag/android/Error.java

```diff
@@ -302,4 +308,48 @@ void setBreadcrumbs(Breadcrumbs breadcrumbs) {
     boolean shouldIgnoreClass() {
         return config.shouldIgnoreClass(getExceptionName());
     }
+
+    static class Builder {
+        private final Configuration config;
+        private final Throwable exception;
+        private Severity severity = Severity.WARNING;
+        private MetaData metaData;
+        private boolean unhandled;
+
+        Builder(@NonNull Configuration config, @NonNull Throwable exception) {
+            this.config = config;
+            this.exception = exception;
+        }
+
+        Builder(@NonNull Configuration config, @NonNull String name,
+               @NonNull String message, @NonNull StackTraceElement[] frames) {
+            this(config, new BugsnagException(name, message, frames));
+        }
+
+        Builder withSeverity(Severity severity) {
+            this.severity = severity;
+            return this;
+        }
+
+        Builder withMetaData(MetaData metaData) {
+            this.metaData = metaData;
+            return this;
+        }
+
+        Builder isUnhandled(boolean unhandled) {
```

Comment:
I'm uncertain about the name of this setter, since it doesn't imply that a value is being set. `withUnhandled`/`withUnhandledValue` are also odd grammatically though 🤔

Reply:
Yep I can see how this could be confusing. How about something like: `builder.severity(WARNING).unhandled(false)`